### PR TITLE
Typescript everything for the API

### PR DESCRIPTION
**Pull request by willvictor**
This PR follows the instructions to use this package: https://www.npmjs.com/package/sequelize-typescript#usage. It provides decorators/apis for using sequelize (the ORM) with typescript. 
It also updates the name of the "domain" folder to just be "database" which I think is clearer.

### CODE REVIEW

File: cookbook-api/database/migrations/RunMigrations.ts [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+    script_name: string;
 }
 
+const basename = path.basename(__filename);
+
+var sequelize = getSequelizeInstance();
 const runDbUp = async () => {
-    const hasScriptsHistoryTable = await sequelize.query(
+
+    const hasScriptsHistoryTable = await sequelize.query<Exists>(
```
Comment (chrisdieckhaus) [inline cookbook-api/database/migrations/RunMigrations.ts:18]: Kind of annoying you have to have a specific `Exists` object just for one boolean field, but I get it.

File: cookbook-api/database/models/Recipe.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import {Table, Column, Model, BelongsTo, CreatedAt, UpdatedAt, ForeignKey} from 'sequelize-typescript';
+import {DataTypes} from 'sequelize';
+import {User} from './User';
+
+@Table({
+    tableName: 'recipes',
+    timestamps: true
+})
+export class Recipe extends Model<Recipe> {
```
Comment (chrisdieckhaus) [inline cookbook-api/database/models/Recipe.ts:9]: What's the difference between `Recipe` and the `Model<Recipe>` it's extending?

File: cookbook-api/database/models/Recipe.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    autoIncrement: true,
+    field: "recipe_id"
+    })
+    recipeId: number;
+
+    @ForeignKey(() => User)
+    @Column({
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        field: "user_id"
```
Comment (chrisdieckhaus) [inline cookbook-api/database/models/Recipe.ts:23]: Feel like `creator_id` is a better name for this field. Makes it clear what the associate of user to recipe is.

**Review by chrisdieckhaus — APPROVED** [review]

File: cookbook-api/database/migrations/RunMigrations.ts (hunk 1 above)
Comment (willvictor) [reply cookbook-api/database/migrations/RunMigrations.ts:18]: agreed- but eh- I still think it's an upside.

File: cookbook-api/database/models/Recipe.ts (hunk 3 above)
Comment (willvictor) [reply cookbook-api/database/models/Recipe.ts]: I agree with that. I've made the change.

In the process- I found a bug I introduced to Dbup by typescript-ifying all of this. 

I think we should add a dbup test to make sure DB up is working correctly. I'm not sure the best way to do that now, so I'll probably just file an issue.

**Pull request merged**